File: Brainstorming/benchmarks/lang-eval/go/taskboard.go
// Package taskboard — identical spec across Python / Go / Rust.
package taskboard

import (
	"errors"
	"fmt"
	"sort"
)

type TaskStatus int

const (
	Todo TaskStatus = iota
	Done
)

// Typed sentinel errors. Go forces an explicit `if err != nil`, but does NOT
// force the caller to actually check the returned error (errcheck is a linter,
// not the compiler).
var (
	ErrEmptyTitle  = errors.New("title must not be empty")
	ErrBadPriority = errors.New("priority must be 1..5")
	ErrNotFound    = errors.New("task not found")
	ErrAlreadyDone = errors.New("task already done")
)

type Task struct {
	ID       int
	Title    string
	Priority int
	Status   TaskStatus
}

type TaskBoard struct {
	tasks  map[int]*Task
	nextID int
}

func New() *TaskBoard {
	return &TaskBoard{tasks: make(map[int]*Task), nextID: 1}
}

func trimSpace(s string) string {
	start, end := 0, len(s)
	for start < end && (s[start] == ' ' || s[start] == '\t' || s[start] == '\n') {
		start++
	}
	for end > start && (s[end-1] == ' ' || s[end-1] == '\t' || s[end-1] == '\n') {
		end--
	}
	return s[start:end]
}

func (b *TaskBoard) Add(title string, priority int) (int, error) {
	t := trimSpace(title)
	if t == "" {
		return 0, ErrEmptyTitle
	}
	if priority < 1 || priority > 5 {
		return 0, fmt.Errorf("%w: got %d", ErrBadPriority, priority)
	}
	task := &Task{ID: b.nextID, Title: t, Priority: priority, Status: Todo}
	b.tasks[task.ID] = task
	b.nextID++
	return task.ID, nil
}

func (b *TaskBoard) Complete(id int) error {
	task, ok := b.tasks[id]
	if !ok {
		return fmt.Errorf("%w: %d", ErrNotFound, id)
	}
	if task.Status == Done {
		return fmt.Errorf("%w: %d", ErrAlreadyDone, id)
	}
	task.Status = Done
	return nil
}

func (b *TaskBoard) Pending() []*Task {
	var todo []*Task
	for _, t := range b.tasks {
		if t.Status == Todo {
			todo = append(todo, t)
		}
	}
	sort.Slice(todo, func(i, j int) bool {
		if todo[i].Priority != todo[j].Priority {
			return todo[i].Priority > todo[j].Priority // highest first
		}
		return todo[i].ID < todo[j].ID
	})
	return todo
}

func (b *TaskBoard) Stats() (total, done, pending int) {
	total = len(b.tasks)
	for _, t := range b.tasks {
		if t.Status == Done {
			done++
		}
	}
	return total, done, total - done
}

File: Brainstorming/benchmarks/lang-eval/python/taskboard.py
"""TaskBoard domain library — identical spec across Python / Go / Rust."""

from dataclasses import dataclass
from enum import Enum


class TaskStatus(Enum):
    TODO = "todo"
    DONE = "done"


class TaskError(Exception):
    """Base domain error. Note: not enforced by the type system — callers may
    forget to catch it (no compile-time signal)."""


class EmptyTitle(TaskError):
    pass


class BadPriority(TaskError):
    def __init__(self, value: int):
        super().__init__(f"priority must be 1..5, got {value}")
        self.value = value


class NotFound(TaskError):
    def __init__(self, task_id: int):
        super().__init__(f"task {task_id} not found")
        self.task_id = task_id


class AlreadyDone(TaskError):
    def __init__(self, task_id: int):
        super().__init__(f"task {task_id} already done")
        self.task_id = task_id


@dataclass
class Task:
    id: int
    title: str
    priority: int
    status: TaskStatus


class TaskBoard:
    def __init__(self) -> None:
        self._tasks: dict[int, Task] = {}
        self._next_id = 1

    def add(self, title: str, priority: int) -> int:
        if not title.strip():
            raise EmptyTitle("title must not be empty")
        if priority < 1 or priority > 5:
            raise BadPriority(priority)
        task = Task(self._next_id, title.strip(), priority, TaskStatus.TODO)
        self._tasks[task.id] = task
        self._next_id += 1
        return task.id

    def complete(self, task_id: int) -> None:
        task = self._tasks.get(task_id)
        if task is None:
            raise NotFound(task_id)
        if task.status is TaskStatus.DONE:
            raise AlreadyDone(task_id)
        task.status = TaskStatus.DONE

    def pending(self) -> list[Task]:
        todo = [t for t in self._tasks.values() if t.status is TaskStatus.TODO]
        # highest priority first, then by id ascending
        return sorted(todo, key=lambda t: (-t.priority, t.id))

    def stats(self) -> tuple[int, int, int]:
        total = len(self._tasks)
        done = sum(1 for t in self._tasks.values() if t.status is TaskStatus.DONE)
        return total, done, total - done
